File: code/2d216.pas
program Bai216;
var a:array[1..1000] of integer;
    i, n: integer;
begin
  readln(n);
  for i := 1 to n do read(a[i]);
  for i := 1 to n do 
    begin
    if (a[i] mod 2) + (a[i+1] mod 2) <> 1 then 
        begin
          writeln ('NO');
          exit;
        end;
    end;
    write('YES');
end.

File: code/2d216.cpp
#include <iostream>
using namespace std;

int main() 
{
    int n,i;
    int a[100000];
    cin>>n;
    for (i=1; i<=n; i++){
        cin>>a[i];
    }
    for (i=1; i<=n-1; i++)
        if (a[i]%2 + a[i+1]%2 != 1){
            cout<<"NO";
            return 0;
        }
    cout<<"YES";
    return 0;
}